docs/extensions/cross-chain-data/oracles.mdx

```diff
@@ -0,0 +1,53 @@
+---
+sidebar_position: 1
+sidebar_label: "Overview"
+id: oracles-overview
+title: Oracles
+description: Build custom event-driven oracles.
+slug: /extensions/oracles-overview
+---
+
+Kwil networks have the ability to come to consensus on external system events, and trigger subsequent state changes based on the result of those events. This allows for network changes to be automatically triggered by external systems, such as other blockchain networks.
+
+Example usecases for Kwil oracles include:
+
+- Giving users native network gas when they burn an ERC20 token on Ethereum.
+- Updating a validator's power when they stake an SPL token on Solana.
+- Automatically syncing data from another Kwil network.
+
+## Anatomy of an Oracle
+
+Oracles are made up of two distinct components: event listeners, and resolutions.
+
+[Event Listeners](/docs/extensions/cross-chain-data/listeners.mdx) are responsible for observing external systems and storing information about the events that occur. When an event listener detects an event, it stores the event in its local database. The node will then broadcast information about the event to the rest of the network, until the event is recognized by a majority of the network.
+
+[Resolutions](/docs/extensions/cross-chain-data/resolutions.mdx) are responsible for taking the information about the event from the event listener, and using it to trigger a state change in the network. When a resolution is triggered, it will broadcast information about the state change to the rest of the network, until the state change is recognized by a majority of the network.
```

Comment:
```suggestion
[**Resolutions**](/docs/extensions/cross-chain-data/resolutions.mdx) are responsible for taking the information about the event from the event listener, and using it to trigger a state change in the network. When a resolution is triggered, it will broadcast information about the state change to the rest of the network, until the state change is recognized by a majority of the network.
```